static/js/recent_topics.js

```diff
@@ -41,20 +40,15 @@ exports.process_message = function (msg) {
         topic_data.starred.add(msg.id);
     }
     topic_data.participated = is_ours || topic_data.participated;
-    topic_data.muted = topic_data.muted || muting.is_topic_muted(msg.stream_id, msg.topic);
     if (initialized) {
         exports.inplace_rerender(key);
     }
     return true;
 };
 
-exports.update_topic_is_muted = function (stream_id, topic, is_muted) {
+exports.toggle_topic_is_muted = function (stream_id, topic) {
     const key = stream_id + ":" + topic;
-    if (!topics.has(key)) {
-        return false;
-    }
-    const topic_data = topics.get(stream_id + ":" + topic);
-    topic_data.muted = is_muted;
+    get_topic_row(key).toggle();
```

Comment:
Are you sure about moving to a `.toggle()` system?  It's harder to read and not safe against getting duplicates of an event...

Reply:
I prefer using e.g. `.toggle(!is_muted)` to just set it to the desired state, rather than having a state machine where we could get a parity error.

Reply:
OK! Switched to this convention.